client/src/components/Atoms/Table.tsx

```diff
@@ -91,22 +93,30 @@ export const Table = <T extends object>({
   data,
   columns,
   id,
+  initialSortBy,
+  onSortByChange,
 }: ITableProps<T>): React.ReactElement => {
   const {
     getTableProps,
     getTableBodyProps,
     headers,
     rows,
     prepareRow,
+    state: { sortBy },
   } = useTable(
     {
       data: React.useMemo(() => data, [data]),
       columns: React.useMemo(() => columns, [columns]),
       autoResetSortBy: false,
+      initialState: initialSortBy && { sortBy: initialSortBy },
     },
     useSortBy
   )
 
+  useEffect(() => {
+    if (onSortByChange) onSortByChange(sortBy)
+  }, [sortBy, onSortByChange])
```

Comment:
This seems to be the recommended way to listen for state changes: https://github.com/TanStack/table/blob/v7/docs/src/pages/docs/faq.md#how-can-i-use-the-table-state-to-fetch-new-data